raiden/claim.py

```diff
@@ -0,0 +1,222 @@
+                encode_single("uint256", self.chain_id)
+                + self.token_network_address
+                + self.partner
+                + self.owner
+            )
+        return ChannelID(int.from_bytes(bytes=hashed_id, byteorder="big"))
+
+
+def parse_claims_file() -> List[Claim]:
+    claims_data = json.loads(CLAIM_FILE_PATH.read_text())
+    return [
+        DictSerializer.deserialize({"_type": "raiden.claim.Claim", **claim})
+        for claim in claims_data["claims"]
+    ]
+
+
+def create_new_token_network_event(
+    token_network_registry_address: TokenNetworkRegistryAddress,
+    token_network_address: TokenNetworkAddress,
+) -> List[StateChange]:
+    return [
+        ContractReceiveNewTokenNetwork(
+            token_network_registry_address=token_network_registry_address,
+            token_network=TokenNetworkState(
+                address=token_network_address,
+                token_address=TOKEN_ADDRESS,
+                network_graph=TokenNetworkGraphState(token_network_address),
+            ),
+            transaction_hash=TransactionHash(b""),
+            block_number=BlockNumber(0),
+            block_hash=BlockHash(b""),
+        ),
+    ]
+
+
+def claims_to_blockchain_events(
+    claims: List[Claim],
+    node_address: Address,
+    token_network_registry_address: TokenNetworkRegistryAddress,
+) -> List[StateChange]:
+    state_changes = create_new_token_network_event(
+        token_network_registry_address, claims[0].token_network_address
+    )
+
+    # TODO: check claim signature
+    for claim in claims:
+        # If node is channel participant, create NettingChannelState
+        if node_address == claim.owner or node_address == claim.partner:
+            our_state = NettingChannelEndState(
+                claim.owner if claim.owner == node_address else claim.partner, Balance(0),
+            )
+            partner_state = NettingChannelEndState(
+                claim.partner if claim.owner == node_address else claim.owner, Balance(0),
+            )
+
+            channel_state = NettingChannelState(
+                canonical_identifier=CanonicalIdentifier(
+                    chain_identifier=claim.chain_id,
+                    token_network_address=claim.token_network_address,
+                    channel_identifier=claim.channel_id(),
+                ),
+                token_address=TOKEN_ADDRESS,
+                token_network_registry_address=token_network_registry_address,
+                reveal_timeout=DEFAULT_REVEAL_TIMEOUT,
+                settle_timeout=DEFAULT_SETTLE_TIMEOUT,
+                fee_schedule=FeeScheduleState(),
+                our_state=our_state,
+                partner_state=partner_state,
+                open_transaction=SuccessfulTransactionState(BlockNumber(0)),
+                close_transaction=None,
+                settle_transaction=None,
+            )
+
+            state_changes.append(
+                ContractReceiveChannelNew(
```

Comment:
I would argue, that we only want to do this once. On the other hand, `raiden.transfer.token_network.handle_channelnew` seems to happily ignore duplicated `channelnew` events.